### PR TITLE
Implement `getCapabilities`

### PR DESCRIPTION
**Pull request by rogurotus**
## Synopsis

Implement `getCapabilities`.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains issue reference
    - [x] Has type and `k::` labels applied
    - [x] Has assignee
- Before [review][l:4]:
    - [x] Documentation is updated (if required)
    - [x] Tests are updated (if required)
    - [x] Changes conform [code style][l:2]
    - [x] [CHANGELOG entry][l:3] is added (if required)
    - [x] FCM (final commit message) is posted or updated
    - [x] [Draft mode][l:1] is removed
- [x] [Review][l:4] is completed and changes are approved
    - [x] FCM (final commit message) is approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: /CONTRIBUTING.md#code-style
[l:3]: /CHANGELOG.md
[l:4]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by rogurotus** [comment]
## FCM

```
Implement getCapabilities (#137)

Additionally:
- implement setCodecPreferences
- upgrade libwebrt to 121.0.6167.160
```

**Review by evdokimovs — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/WebrtcVideoEncoderFactory.kt [hunk 1]
```diff
@@ -46,6 +47,10 @@ class WebrtcVideoEncoderFactory
     return codecs.toTypedArray()
   }
 
+  override fun queryCodecSupport(codecInfo: VideoCodecInfo, scalability: String): CodecSupport {
+    return swFactory.queryCodecSupport(codecInfo, scalability)
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/WebrtcVideoEncoderFactory.kt:55]: Fixed

File: crates/libwebrtc-sys/src/bridge.rs [hunk 2]
```diff
@@ -653,6 +653,208 @@ pub(crate) mod webrtc {
...
+
+    #[derive(Debug, Eq, Hash, PartialEq)]
+    #[repr(i32)]
+    // Used in RtcpFeedback struct.
+    pub enum RtcpFeedbackType {
+        CCM,
+        LNTF, // "goog-lntf"
+        NACK,
+        REMB, // "goog-remb"
+        TRANSPORT_CC,
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply crates/libwebrtc-sys/src/bridge.rs]: Fixed

File: crates/libwebrtc-sys/src/lib.rs [hunk 3]
```diff
@@ -893,6 +905,165 @@ impl RtpTransceiverInterface {
     }
 }
 
+/// `NACK` or `CCM` feedback.
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply crates/libwebrtc-sys/src/lib.rs]: Fixed

File: crates/libwebrtc-sys/src/lib.rs [hunk 4]
```diff
@@ -893,6 +905,165 @@ impl RtpTransceiverInterface {
...
+    /// Returns the `message_type` of these [`RtcpFeedback`].
+    ///
+    /// # Panics
+    ///
+    /// If [`RtcpFeedbackMessageType`] has invalid type.
+    #[must_use]
+    pub fn message_type(&self) -> Option<webrtc::RtcpFeedbackMessageType> {
+        webrtc::rtcp_feedback_message_type(&self.0)
+            .take()
+            .map(|v| webrtc::RtcpFeedbackMessageType::try_from(v).unwrap())
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply crates/libwebrtc-sys/src/lib.rs]: Wrapped in `OptionRtcpFeedbackMessageType`. It is checked at the compilation stage

File: crates/native/src/api.rs [hunk 5]
```diff
@@ -516,6 +516,416 @@ impl From<sys::IceCandidateStats> for IceCandidateStats {
...
+        match value {
+            sys::RtcpFeedbackMessageType::GENERIC_NACK => Self::GenericNACK,
+            sys::RtcpFeedbackMessageType::PLI => Self::PLI,
+            sys::RtcpFeedbackMessageType::FIR => Self::FIR,
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// NACK or CCM feedback.
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply crates/native/src/api.rs]: Fixed

File: crates/native/src/api.rs [hunk 6]
```diff
@@ -516,6 +516,416 @@ impl From<sys::IceCandidateStats> for IceCandidateStats {
...
+    /// [ScalabilityMode.S3T3h][0] representation.
+    ///
+    /// [0]: https://www.w3.org/TR/webrtc-svc/#S3T3*
+    S3T3h,
+}
+
+#[derive(Debug, Eq, Hash, PartialEq)]
+#[repr(i32)]
+// Used in RtcpFeedback struct.
+#[allow(clippy::upper_case_acronyms)]
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply crates/native/src/api.rs]: Fixed

File: lib/src/api/transceiver.dart [hunk 7]
```diff
@@ -173,6 +184,26 @@ class _RtpTransceiverFFI extends RtpTransceiver {
         direction: ffi.RtpTransceiverDirection.values[direction.index]);
   }
 
+  // todo
```
[the remark this answers is not in the document]
Comment (rogurotus) [reply lib/src/api/transceiver.dart]: Deleted

**Review by evdokimovs — APPROVED** [review]

**Review by tyranron — APPROVED** [review]
@rogurotus @evdokimovs please do not forget CHANGELOG next time (you have that in checklist!) and be more pedantic regarding docs.

**Pull request merged**